### ISSUE TITLE
some problems when install torchvision from source offline

### ISSUE THREAD

**Issue by shaoming20798** [pytorch/vision#2632]
Hi, I have installed pytorch from source following the instruction in https://github.com/pytorch/pytorch. Then, I installed torchvison also from source, using this instruction "python setup.py install". And I got this message
> Installed /opt/conda/envs/pytorch_src/lib/python3.8/site-packages/torchvision-0.8.0a0+10d5a55-py3.8-linux-x86_64.egg
Processing dependencies for torchvision==0.8.0a0+10d5a55
Searching for torch
Reading https://pypi.org/simple/torch/
Downloading https://files.pythonhosted.org/packages/8c/5d/faf0d8ac260c7f1eda7d063001c137da5223be1c137658384d2d45dcd0d5/torch-1.6.0-cp38-cp38-manylinux1_x86_64.whl#sha256=5357873e243bcfa804c32dc341f564e9a4c12addfc9baae4ee857fcc09a0a216

I cannot connect to the internet. So what should I do next? Please give me a help. thanks.

**Comment by fmassa** [comment]
Hi,

I think there might be a conflict of versions with setuptools due to the fact that you installed pytorch from source.

You might be able to pass the environment variable https://github.com/pytorch/vision/blob/497744b9d510ff2df756f479ee5a19fce0d579b6/setup.py#L65-L66
to set to the commit hash that you are using for pytorch, although I'm not sure this is going to solve your issue.

Tagging @seemethere for ideas on how to solve this. I have never encountered this issue, but I also never compile PyTorch from source.

**Comment by seemethere** [comment]
This shouldn't really be an issue,

If you don't specify a `PYTORCH_VERSION` then the dependency should just be `torch`.

Are you sure you installed torch correctly? What happens when you do `python -c 'import torch'`?

**Comment by shaoming20798** [comment]
> This shouldn't really be an issue,
> 
> If you don't specify a `PYTORCH_VERSION` then the dependency should just be `torch`.
> 
> Are you sure you installed torch correctly? What happens when you do `python -c 'import torch'`?

Yes, I'm sure I have installed pytorch correctly, because I have trained a model correctly using that installed pytorch and it showed the right version when I do 'import torch; print(torch.__version__)' in my train.py file.
I have installed torchvision successfully from source using `python setup.py install`. But torch-1.6.0 have been forced installed in my computer through downloading torch-1.6.0 though I have installed pytorch. 
Did I make some mistake?

**Comment by fmassa** [comment]
Could it be that there are multiple python environments and you might be using different ones by chance?

**Comment by ksasso1028** [comment]
I am having a similar issue where I cannot pass any env variables to the torchvision setup.py script. I am using a venv and have compiled torch from source using the same venv which works. My CUDA and torch locations are not in the default usr dirs and cannot get torchvision to find my cuda install without directly editing setup.py which still fails for nvcc

**Comment by felker** [comment]
Chiming in to report a similar issue when PyTorch is built from source.

```
pip list | grep -i torch
torch                             1.9.0a0+gitd69c22d
```
which is installed from `torch-1.9.0a0+gitd69c22d-cp38-cp38-linux_x86_64.whl`, a wheel I built from source and is working totally fine on a DGX machine. 

`PYTORCH_VERSION` is unset. When I try `pip install torchvision`, pip attempts to install a new PyTorch 1.9.0 wheel. 

@seemethere any thoughts on how to debug the logic of `setup.py`? I agree that it seems like it shouldn't be trying to do this if the dependency is only unversioned `torch`.

**Comment by fmassa** [comment]
@felker we only provide torchvision binaries for either the nightlies of PyTorch, or for PyTorch releases.

This means that if you compile PyTorch from source you'll also need to compile torchvision from source, by doing `python setup.py install` in the torchvision repo.
